pmlc/rt/opencl/opencl_runtime.cc

```diff
@@ -29,6 +30,13 @@ class OpenCLRuntime final : public pmlc::rt::Runtime {
   std::vector<std::shared_ptr<OpenCLDevice>> devices;
 };
 
-pmlc::rt::RuntimeRegistration<OpenCLRuntime> reg{"opencl"};
+extern void registerSymbols();
+extern void registerMemrefCasts();
+
+void registerRuntime() {
+  registerSymbols();
+  registerMemrefCasts();
+  registerRuntime("opencl", std::make_shared<OpenCLRuntime>());
```

Comment:
Might as well try/catch here too.

Reply:
Ok let me change both vulkan and ocl to what you proposed